### PR TITLE
feat: handle if/block/match nodes in system

### PR DESCRIPTION
**Pull request by broody**
Resolves https://github.com/dojoengine/dojo/issues/156

### CODE REVIEW

**Comment by ponderingdemocritus** [comment]
nice work!

File: crates/dojo-lang/src/plugin_test_data/system [hunk 1]
```diff
@@ -88,6 +88,35 @@ mod Move {
...
+            }
+        } else if bar == 321 {
+            let player_position = commands::<(Position, Player)>::get(player_id.into());
+        } else {
+            let player_position = commands::<(Position, Player)>::get(player_id.into());
+        }
+
+        {
+            let player_position = commands::<(Position, Player)>::get(player_id.into());
+        }
```
Comment (tarrencev) [inline crates/dojo-lang/src/plugin_test_data/system:104]: oh interesting i didn't know this was a valid expression

**Review by tarrencev — APPROVED** [review]

File: crates/dojo-lang/src/plugin_test_data/system [hunk 2]
```diff
@@ -454,15 +483,131 @@ mod MoveSystem {
...
+        let __player_position_position = serde::Serde::<Position>::deserialize(
+            ref __player_position_position_raw
+        );
+
+        let mut __player_position_player_raw = IWorldDispatcher {
+            contract_address: world_address
+        }.get('Player', player_id.into(), 0_u8, 0_usize);
+        let __player_position_player = serde::Serde::<Player>::deserialize(
+            ref __player_position_player_raw
+        );
```
Comment (tarrencev) [inline crates/dojo-lang/src/plugin_test_data/system:554]: it looks like we lost the scoped block here

Comment (broody) [reply crates/dojo-lang/src/plugin_test_data/system:554]: Ah nice catch

**Pull request merged**